src/devices/tuya.ts

```diff
@@ -2604,6 +2657,31 @@ export const definitions: DefinitionWithExtend[] = [
         },
         whiteLabel: [tuya.whitelabel("GIEX", "GX04", "Soil Moisture Sensor", ["_TZE284_nhgdf6qr"])],
     },
+    {
+        fingerprint: tuya.fingerprint("SGS02Z", ["_TZE284_nt4pquef"]),
+        model: "SGS02Z",
+        vendor: "Tuya",
+        description: "Soil sensor",
+        fromZigbee: [fzLocal.sgs02zUnitConvert],
```

Comment:
please use tuya.fz.datapoints here, any custom conversion logic can be done in a [valueConverter](https://github.com/Koenkk/zigbee-herdsman-converters/blob/f6551ae677dc620f56eb30271341aa202ecc6f27/src/lib/tuya.ts#L499)